test/e2e/prometheusadapter_test.go

```diff
@@ -53,43 +53,81 @@ func TestMetricsAPIAvailability(t *testing.T) {
 		if err == wait.ErrWaitTimeout && lastErr != nil {
 			err = lastErr
 		}
-		log.Fatal(err)
+		t.Fatal(err)
 	}
 }
 
 func TestNodeMetricsPresence(t *testing.T) {
 	var lastErr error
 	err := wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		nodes, err := f.KubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
+		lastErr = errors.Wrap(err, "getting nodes list failed")
+		if err != nil {
+			return false, nil
+		}
 		nodeMetrics, err := f.MetricsClient.MetricsV1beta1().NodeMetricses().List(metav1.ListOptions{})
 		lastErr = errors.Wrap(err, "getting metrics list failed")
 		if err != nil {
 			return false, nil
 		}
+		if len(nodes.Items) != len(nodeMetrics.Items) {
+			lastErr = errors.New("number of nodes doesn't match number of node metrics reported")
+			return false, nil
+		}
 		for _, item := range nodeMetrics.Items {
+			for i, node := range nodes.Items {
+				if node.Name == item.Name {
+					break
+				}
+				if i >= len(nodes.Items) {
```

Comment:
Not sure this condition is correct. Let's extract "checkIfMetricsExist" to a separate function, then this logic should become a little simpler to read.

Reply:
I have extracted checking if pod/node is in pods/nodes list. This should increase readability a bit.